multiqc/modules/sambamba/markdup.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+from collections import OrderedDict
+import os
+import logging
+import re
+
+from multiqc import config
+from multiqc.plots import bargraph
+
+log = logging.getLogger(__name__)
+
+
+class SambambaMarkdupMixin:
+
+    """Find and parse Sambamba Markdup output log files and calculate duplication rate"""
+
+    def parse_sambamba_markdup(self):
+
+        # Clean sample name from 'markdup_sample_1' to 'sample_1'
+        # Find and load sambamba logs to markdup_data_dict.
+        # Regex for key phrases and calculate duplicate rate.
+        # Give user warning if redundant samples are found.
+
+        self.markdup_data_dict = dict()
+
+        for f in self.find_log_files("sambamba/markdup"):
+
+            # define sample name
+            sample_name = self.clean_s_name(f["s_name"].replace("markdup_", ""), f["root"])

```

Comment:
This requires the user to specify a log filename that matches this format, right? It's not a standardised log filename from the tool itself?

I generally try to avoid doing this if at all possible, as users sometimes don't have any choice about the filenames that are generated to capture this kind of output.

Having said this, the log here looks pretty horrible with no mention of input filename. Maybe we just take `f["s_name"]` and use that, whatever it is. We should definitely remove the `replace` bit though and use the central MultiQC sample name cleaning configs to do this for us.

Reply:
Ah ignore me, I already suggested that and then remembered that it's suffixes only for the config cleaning: https://github.com/ewels/MultiQC/pull/1417#discussion_r622517152

Sorry! But yeah, I think that we can probably still remove this completely if the filename convention is just something you're doing yourself manually and not coming from the tool.

Reply:
For Sambamba Markdup, the user can freely name the log file. I removed it completely so now it's just whatever the user puts down: 

`sample_name = f["s_name"]`

This may ultimately come down to preference, but I'm wondering if it's easier to just drop the `markdup_` prefix in general? This format is not a standard by any means. I think dropping the prefix (and relying on the content search pattern) will better allow the end user to use and even "discover" this module more easily. Does this make sense?